src/clj_btc/json_rpc.clj

```diff
@@ -51,7 +53,15 @@
                                   "method" method,
                                   "id" (.incrementAndGet id-num)})})]
     (if-let [err (:error resp)]
-      resp
-      (if (= 200 (:status resp))
-        (-> resp :body StringReader. (json/read :bigdec true) (get "result"))
-        (-> resp :body json/read-str (get "error"))))))
+      (do
+        (if (fn? logger)
```

Comment:
sorry for being a pain: `(when (fn? logger)` is the prefered way ([style guide](https://github.com/bbatsov/clojure-style-guide#when-instead-of-single-branch-if))